chore: Pass cron schedule to workflow (#5138)
argoproj/argo-workflows: docs/variables.md

| `workflow.creationTimestamp.<STRFTIMECHAR>` | Creation timestamp formatted with a [strftime](http://strftime.org) format character |
| `workflow.priority` | Workflow priority |
| `workflow.duration` | Workflow duration estimate, may differ from actual duration by a couple of seconds |

> | `workflow.parameters.cronScheduleTime` | Cron workflow Scheduled timestamp formatted in RFC 3339

Reviewer: You should specify this is only available for cron workflows
```suggestion
| `workflow.parameters.cronScheduleTime` | Scheduled runtime formatted in RFC 3339 (only available for CronWorkflows) |
```

> | `workflow.parameters.scheduledTime` | Scheduled runtime formatted in RFC 3339 (only available for CronWorkflows) |

Reviewer 2: this collides with any parameter with the same name, how about just `workflow.scheduledTime`?

Author: yes maybe. That's why initially I gave more specific `cronScheduledTime` if you want, we have `cronwfScheduleTime` or your suggession

It has to be a parameter so it will work flawlessly with our existing parameter logic. Current `workflow.x` variables are calculated during runtime or there is a field in `spec`. If we need to have `workflow.scheduledTime`, we need to add new field in `spec`

Reviewer: I don't think it's too unreasonable to reserve a parameter name much like programming languages reserve keywords. But I agree with Alex that ideally we'd use `workflow.scheduledTime` or even `cronWorkflow.scheduledTime`

Reviewer: Although at this point introducing `workflow.parameters.scheduledTime` might break users who already use that parameter name

Reviewer 2: I think `scheduledTime` is fine

Author: @alexec and @simster7 Agreed. Let me go with `workflow.scheduledTime`

Reviewer: I actually think `cronWorkflow.scheduledTime` is better... any opposition to that?

Author: @simster7  I would suggest sticking with `workflow.scheduledTime`, so code doesn't need to check another string for global variable

Reviewer: Hmm, thinking about it more I think `workflow.` is right... it's the Workflow that got scheduled at this particular time, not the CronWorkflow
